core/src/main/java/org/apache/druid/java/util/common/parsers/ObjectFlatteners.java

```diff
@@ -218,9 +219,9 @@ public Map<String, Object> toMap(T obj)
   {
     JsonProvider getJsonProvider();
     /**
-     * List all "root" primitive properties and primitive lists (no nested objects, no lists of objects)
+     * List all "root" fields, optionally filtering to include only fields that contain primitive and lists of primitive values
      */
-    Iterable<String> discoverRootFields(T obj);
+    Iterable<String> discoverRootFields(T obj, boolean discoverNestedFields);
```

Comment:
I don't think you need to change the interface at all?  Can't you add the config as a property on the actual FlattenerMaker objects themselves?  Doing that will definitely shrink this PR down and avoid interface changes?

Reply:
yeah, i guess I was thinking doing it this way would force implementors to be aware that the contract is a bit different (or at least will be), though maybe it is nicer to push into the `FlattenerMaker` constructors instead and we just mention in the release notes that implementors should try to honor the mode for their readers.